1.
По данным опроса, 75% работников ресторанов утверждают, что испытывают на работе существенный стресс, оказывающий негативное влияние на их личную жизнь. Крупная ресторанная сеть опрашивает 100 своих работников, чтобы выяснить, отличается ли уровень стресса работников в их ресторанах от среднего. 67 из 100 работников отметили высокий уровень стресса.

Посчитайте достигаемый уровень значимости, округлите ответ до четырёх знаков после десятичной точки.

In [2]:
import numpy as np
from scipy import stats
import pandas as pd

In [3]:
round(stats.binom_test(67, 100, 0.75), 4)

0.0822

answer1 = 0.0822

2.
Представим теперь, что в другой ресторанной сети только 22 из 50 работников испытывают существенный стресс. Гипотеза о том, что 22/50 соответствует 75% по всей популяции, методом, который вы использовали в предыдущей задаче, отвергается. Чем это может объясняться? Выберите все возможные варианты.

In [7]:
stats.binom_test(22, 50, 0.75)

3.4831478978300243e-06



Условия работы в этой сети лучше, чем в среднем


Сеть известна низкими зарплатами и большой текучкой кадров, поэтому туда идут работать люди, не склонные переживать о сохранении рабочего места


Эта сеть менее популярна, поэтому в её ресторанах меньше поток посетителей; отсюда меньший уровень стресса у работников


Сотрудники чрезмерно оптимистично отвечали на вопрос, потому что опрос не был анонимным



3.
The Wage Tract — заповедник в округе Тома, Джорджия, США, деревья в котором не затронуты деятельностью человека со времён первых поселенцев. Для участка заповедника размером 200х200 м имеется информация о координатах сосен (sn — координата в направлении север-юг, we — в направлении запад-восток, обе от 0 до 200).

Проверим, можно ли пространственное распределение сосен считать равномерным, или они растут кластерами.

Загрузите данные, поделите участок на 5х5 одинаковых квадратов размера 40x40 м, посчитайте количество сосен в каждом квадрате (чтобы получить такой же результат, как у нас, используйте функцию scipy.stats.binned_statistic_2d).

Если сосны действительно растут равномерно, какое среднее ожидаемое количество сосен в каждом квадрате? В правильном ответе два знака после десятичной точки.

In [10]:
data = pd.read_csv('pines.txt', header = 0, sep = '\\t')

<ipython-input-10-8f652c2b6e1e>:1: ParserWarning: Falling back to the 'python' engine because the 'c' engine does not support regex separators (separators > 1 char and different from '\s+' are interpreted as regex); you can avoid this warning by specifying engine='python'.
  data = pd.read_csv('pines.txt', header = 0, sep = '\\t')


In [11]:
data

sn     we
0    200.0    8.8
1    199.3   10.0
2    193.6   22.4
3    167.7   35.6
4    183.9   45.4
..     ...    ...
579  169.1  160.0
580  162.5  157.3
581  156.7  155.3
582  154.1  150.8
583   87.7  200.0

[584 rows x 2 columns]

In [12]:
data.columns = ['sn', 'we']

In [13]:
X = data.sn.values
Y = data.we.values

In [14]:
binx = [0.0, 40., 80., 120., 160., 200.]
biny = [0.0, 40., 80., 120., 160., 200.]

In [16]:
ret = stats.binned_statistic_2d(X, Y, None, statistic  = 'count', bins=[binx,biny])
ret.statistic

array([[18., 26., 29., 25., 20.],
       [14., 12., 22., 34., 25.],
       [12., 24., 15., 50., 37.],
       [ 8., 19., 31., 50.,  7.],
       [ 7.,  8., 37., 27., 27.]])

In [17]:
l = 584/25
l

23.36

Answer3 = 23.36

4.
Чтобы сравнить распределение сосен с равномерным, посчитайте значение статистики хи-квадрат для полученных 5х5 квадратов. Округлите ответ до двух знаков после десятичной точки.

In [18]:
observed_frequences = np.concatenate((ret.statistic[0], ret.statistic[1], ret.statistic[2], 
                                      ret.statistic[3], ret.statistic[4]), axis=0)
observed_frequences

array([18., 26., 29., 25., 20., 14., 12., 22., 34., 25., 12., 24., 15.,
       50., 37.,  8., 19., 31., 50.,  7.,  7.,  8., 37., 27., 27.])

In [19]:
chi = stats.chisquare(observed_frequences, ddof = 0)
chi

Power_divergenceResult(statistic=150.58904109589042, pvalue=2.574669774967279e-20)

In [20]:
round(chi[0], 2)

150.59

Answer4 = 150.59

5.
Насколько велико это значение? Если нулевая гипотеза справедлива, с какой вероятностью его можно было получить случайно?

Нулевое распределение статистики — хи-квадрат с 25-1=2425−1=24 степенями свободы (поскольку у равномерного распределения, с которым мы сравниваем данные, нет ни одного оцениваемого по выборке параметра, число степеней свободы K-1K−1, где KK — количество интервалов).

Посчитайте достигаемый уровень значимости.

Если вы используете функцию scipy.stats.chi2.cdf, в качестве значения параметра df нужно взять 24 (это число степеней свободы); если функцию scipy.stats.chisquare — параметр ddof нужно брать равным 0 (это как раз количество параметров теоретического распределения, оцениваемых по выборке).

Отвергается ли гипотеза равномерности на уровне значимости 0.05?


Достигаемый уровень значимости больше 0.05, гипотеза равномерности не отвергается.


#### Достигаемый уровень значимости меньше 0.05, гипотеза равномерности отвергается.



In [23]:
chi[1]

2.574669774967279e-20